---
title: "JGI CSP metaT sample selection"
output:
  html_document: 
    fig_width: 12
---


```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here("~/projects/glamr_queries_and_analyses"))
library(tidyverse)
library(googlesheets4)
library(googledrive)
library(DBI)

conflicted::conflicts_prefer(dplyr::select(), dplyr::rename(), dplyr::filter(), dplyr::lag(),base::as.data.frame())

# Open database connection
pg <- DBI::dbConnect(RPostgres::Postgres(),dbname = "glamr_data", host = "cayman.earth.lsa.umich.edu", port = "5432", user = "glamr_admin", password = "glamr2023")
```


```{r, include=FALSE}

options(gargle_oath_cache = "~/GLAMR/.secrets",
        gargle_oauth_email = "kiledal@umich.edu",
        use_oob = TRUE,
        gargle_oauth_client_type = "web")

requested_samples <- googlesheets4::read_sheet("https://docs.google.com/spreadsheets/d/1EEx7MrQ0Etqqmx0HNvP6We7u3MumdbKs7WQK743HxC8/edit?gid=1556500356#gid=1556500356",sheet = "metaT_requests",range = "A1:H1000") %>% 
  filter(!is.na(glamr_metagenome_id))
```

```{r, include=FALSE}
glamr_pg <- tbl(pg, "glamr_samples")

set_35_samples <- glamr_pg %>% 
  filter(StudyID == "set_35") %>% 
  collect() %>% 
  write_tsv("~/projects/glamr_queries_and_analyses/202501_jgi_csp_transcriptomes/set_35_metagenomes.tsv")

microcystis_abundance <- tbl(pg, "tax_abund_from_contig_lca_and_abund") %>% 
  filter(tax_id == 1125,sample %in% local(set_35_samples$SampleID)) %>% 
  mutate(percent_abundance = (abund_w_subtax / 1000000)*100) %>% 
  collect() %>% 
  mutate(microcystis_abundance = round(percent_abundance,4),
         norm_microcystis_abundance = scale(log(microcystis_abundance)))

nostocales_abundance <- tbl(pg, "tax_abund_from_contig_lca_and_abund") %>% 
  filter(tax_id == 1161,sample %in% local(set_35_samples$SampleID)) %>% 
  mutate(percent_abundance = (abund_w_subtax / 1000000)*100) %>% 
  collect() %>% 
  mutate(nostocales_abundance = round(percent_abundance,4))

sites_w_most_samples <- set_35_samples %>% 
  group_by(NOAA_Site) %>% 
  summarise(n_samples = n()) %>% 
  arrange(desc(n_samples))


samples_for_plot <- set_35_samples %>% 
  left_join(requested_samples %>% dplyr::select(SampleID = "glamr_metagenome_id", selection_reason, include, requested_by, notes)) %>% 
  mutate(include = if_else(is.na(include), "no", include)) %>% 
  filter(!is.na(year)) %>% 
  left_join(microcystis_abundance %>% select(SampleID = "sample", microcystis_abundance, norm_microcystis_abundance)) %>% 
  left_join(nostocales_abundance %>% select(SampleID = "sample", nostocales_abundance))
```


```{r, include = FALSE}
bloom_metadata <- read_tsv("~/projects/microcystis_genomes_2022/data/env_data/merged_env_data_GLAMR.tsv")
```
## Bloom summary normalized by year
```{r echo=FALSE,message=FALSE, warning=FALSE}
(
  bloom_metadata %>% 
  mutate(year = lubridate::year(collection_date),
         date_same_year = lubridate::`year<-`(collection_date, 2000)) %>% 
  select(date, year, date_same_year, collection_date, NOAA_site, chlorophyl, ext_phyco, part_microcyst) %>% 
  pivot_longer(c(chlorophyl, ext_phyco, part_microcyst), names_to = "value_type",values_to = "value") %>% 
  group_by(value_type, year) %>% 
  mutate(value = scale(value)) %>% 
  filter(year > 2014, !is.na(year), !is.na(date_same_year)) %>% 
  ggplot(aes(date_same_year, value, 
             color = value_type
             )) +
  geom_smooth(se = FALSE) +
   geom_point(data = samples_for_plot,
              aes(date_same_year, norm_microcystis_abundance, 
                  sample = SampleID, 
                  date = date, 
                  site = NOAA_Site, 
                  color = include, 
                  include_reason = selection_reason, 
                  who = requested_by, 
                  microcystis_percent = microcystis_abundance), inherit.aes = FALSE, size = 2.5) +
    scale_color_manual(values = c("green", "blue", "cyan","grey50", "red", "darkgreen")) + 
  facet_grid(year ~ fct_relevel(NOAA_Site,sites_w_most_samples$NOAA_Site),scales = "free_x") +
  theme_bw() +
  labs(x = NULL, y= NULL) +
    theme(axis.text.y = element_blank(),
          axis.ticks.y = element_blank())
) %>% plotly::ggplotly()
```

## Bloom summary normalized across all years
```{r echo=FALSE, message=FALSE, warning=FALSE}
(
  bloom_metadata %>% 
  mutate(year = lubridate::year(collection_date),
         date_same_year = lubridate::`year<-`(collection_date, 2000)) %>% 
  select(date, year, date_same_year, collection_date, NOAA_site, chlorophyl, ext_phyco, part_microcyst) %>% 
  pivot_longer(c(chlorophyl, ext_phyco, part_microcyst), names_to = "value_type",values_to = "value") %>% 
  group_by(value_type) %>% 
  mutate(value = scale(value)) %>% 
  filter(year > 2014, !is.na(year), !is.na(date_same_year)) %>% 
  ggplot(aes(date_same_year, value, 
             color = value_type
             )) +
  geom_smooth(se = FALSE) +
   geom_point(data = samples_for_plot,
              aes(date_same_year, norm_microcystis_abundance, 
                  sample = SampleID, 
                  date = date, 
                  site = NOAA_Site, 
                  color = include, 
                  include_reason = selection_reason, 
                  who = requested_by, 
                  microcystis_percent = microcystis_abundance), inherit.aes = FALSE, size = 2.5) +
    scale_color_manual(values = c("green", "blue", "cyan","grey50", "red", "darkgreen")) + 
  facet_grid(year ~ fct_relevel(NOAA_Site,sites_w_most_samples$NOAA_Site),scales = "free_x") +
  theme_bw() +
  labs(x = NULL, y= NULL) +
    theme(axis.text.y = element_blank(),
          axis.ticks.y = element_blank())
) %>% plotly::ggplotly()
```






