---
title: "JGI CSP metaT sample selection"
output:
  html_document: 
    fig_width: 12
---


```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here("~/projects/202501_CSP_metaT_selection"))
library(tidyverse)
library(googlesheets4)
library(googledrive)
library(DBI)

conflicted::conflicts_prefer(dplyr::select(), dplyr::rename(), dplyr::filter(), dplyr::lag(),base::as.data.frame())

# Open database connection
pg <- DBI::dbConnect(RPostgres::Postgres(),dbname = "glamr_data", host = "cayman.earth.lsa.umich.edu", port = "5432", user = "glamr_admin", password = "glamr2023")
```


```{r, include=FALSE}

options(gargle_oath_cache = "~/GLAMR/.secrets",
        gargle_oauth_email = "kiledal@umich.edu",
        use_oob = TRUE,
        gargle_oauth_client_type = "web")

requested_samples <- googlesheets4::read_sheet("https://docs.google.com/spreadsheets/d/1EEx7MrQ0Etqqmx0HNvP6We7u3MumdbKs7WQK743HxC8/edit?gid=1556500356#gid=1556500356",sheet = "metaT_requests",range = "A1:H1000") %>% 
  filter(!is.na(glamr_metagenome_id)) %>% 
  write_tsv("requested_samples.tsv")
```

```{r, include=FALSE}
glamr_pg <- tbl(pg, "glamr_samples")

set_35_samples <- glamr_pg %>% 
  filter(StudyID == "set_35") %>% 
  collect() %>% 
  mutate(NOAA_Site = case_when(NOAA_Site == "WE12-S" ~"WE12",
                               NOAA_Site == "WE16-WSW" ~"WE16",
                               .default = NOAA_Site)) %>% 
  write_tsv("~/projects/glamr_queries_and_analyses/202501_jgi_csp_transcriptomes/set_35_metagenomes.tsv")

microcystis_abundance <- tbl(pg, "tax_abund_from_contig_lca_and_abund") %>% 
  filter(tax_id == 1125,sample %in% local(set_35_samples$SampleID)) %>% 
  mutate(percent_abundance = (abund_w_subtax / 1000000)*100) %>% 
  collect() %>% 
  mutate(microcystis_abundance = round(percent_abundance,4),
         norm_microcystis_abundance = scales::rescale(log(microcystis_abundance),to=c(0,1)))

nostocales_abundance <- tbl(pg, "tax_abund_from_contig_lca_and_abund") %>% 
  filter(tax_id == 1161,sample %in% local(set_35_samples$SampleID)) %>% 
  mutate(percent_abundance = (abund_w_subtax / 1000000)*100) %>% 
  collect() %>% 
  mutate(nostocales_abundance = round(percent_abundance,4))

sites_w_most_samples <- set_35_samples %>% 
  group_by(NOAA_Site) %>% 
  summarise(n_samples = n()) %>% 
  arrange(desc(n_samples))


samples_for_plot <- set_35_samples %>% 
  left_join(requested_samples %>% dplyr::select(SampleID = "glamr_metagenome_id", selection_reason, include, requested_by, notes)) %>% 
  mutate(include = if_else(is.na(include), "no", include)) %>% 
  filter(!is.na(year)) %>% 
  left_join(microcystis_abundance %>% select(SampleID = "sample", microcystis_abundance, norm_microcystis_abundance)) %>% 
  left_join(nostocales_abundance %>% select(SampleID = "sample", nostocales_abundance))
```


```{r, include = FALSE}
bloom_metadata <- read_tsv("~/projects/microcystis_genomes_2022/data/env_data/merged_env_data_GLAMR.tsv") %>% 
  dplyr::rename(NOAA_Site = "NOAA_site")
```

## Bloom summary normalized across all years
Key bloom parameters (chlorophyll, phycocyanin, and particulate microcystin) from the complete weekly monitoring dataset are shown to approximate bloom characteristics for each year. 

In this case, the parameter values were scaled across all years to obtain values on a similar scale for each parameter and highlight differences between bloom years. The y-axis position of sample points corresponds to Microcystis abundance in the metagenomes.
```{r echo=FALSE, message=FALSE, warning=FALSE}
(
  bloom_metadata %>% 
  mutate(year = lubridate::year(collection_date),
         date_same_year = lubridate::`year<-`(collection_date, 2000)) %>% 
  select(date, year, date_same_year, collection_date, NOAA_Site, chlorophyl, ext_phyco, part_microcyst,Nitrate_Nitrite) %>% 
  pivot_longer(c(chlorophyl, ext_phyco, part_microcyst,Nitrate_Nitrite), names_to = "value_type",values_to = "value") %>% 
  group_by(value_type) %>% 
  mutate(value = scales::rescale(x= log(value),to=c(0,1))) %>% 
  filter(year > 2014, !is.na(year), !is.na(date_same_year), NOAA_Site %in% samples_for_plot$NOAA_Site) %>% 
  ggplot(aes(date_same_year, value, 
             color = value_type
             )) +
    #geom_smooth(se = FALSE,method = "loess", span = 0.5) +
    geom_path() +
   geom_point(data = samples_for_plot,
              aes(date_same_year, norm_microcystis_abundance, 
                  sample = SampleID, 
                  orig_name = SampleName,
                  date = date, 
                  site = NOAA_Site, 
                  color = include, 
                  include_reason = selection_reason, 
                  who = requested_by, 
                  microcystis_percent = microcystis_abundance), inherit.aes = FALSE, size = 2.5) +
    scale_color_manual(values = c("green", "blue", "#edbb47","pink","#6c5a8f", "red", "#46874a")) + 
  facet_grid(year ~ fct_relevel(NOAA_Site,sites_w_most_samples$NOAA_Site),scales = "free_x") +
  theme_bw() +
  labs(x = NULL, y= "log values") +
    theme(axis.text.y = element_blank(),
          axis.ticks.y = element_blank()) +
    coord_cartesian(ylim = c(0,1))
) %>% plotly::ggplotly()
```


## Bloom summary normalized by year
Key bloom parameters (chlorophyll, phycocyanin, and particulate microcystin) from the complete weekly monitoring dataset are shown to approximate bloom characteristics for each year. 

In this case, the parameter values were scaled within years to obtain values on a similar scale and highlight differences throughout each bloom.  The y-axis position of sample points corresponds to Microcystis abundance in the metagenomes.
```{r echo=FALSE,message=FALSE, warning=FALSE}
(
  bloom_metadata %>% 
  mutate(year = lubridate::year(collection_date),
         date_same_year = lubridate::`year<-`(collection_date, 2000)) %>% 
  select(date, year, date_same_year, collection_date, NOAA_Site, chlorophyl, ext_phyco, part_microcyst,Nitrate_Nitrite) %>% 
  pivot_longer(c(chlorophyl, ext_phyco, part_microcyst,Nitrate_Nitrite), names_to = "value_type",values_to = "value") %>% 
  group_by(value_type, year) %>% 
  mutate(value = scales::rescale(x= log(value),to=c(0,1))) %>% 
  filter(year > 2014, !is.na(year), !is.na(date_same_year), NOAA_Site %in% samples_for_plot$NOAA_Site) %>% 
  ggplot(aes(date_same_year, value, 
             color = value_type
             )) +
  #geom_smooth(se = FALSE) +
    geom_path() +
   geom_point(data = samples_for_plot,
              aes(date_same_year, norm_microcystis_abundance, 
                  sample = SampleID, 
                  orig_name = SampleName,
                  date = date, 
                  site = NOAA_Site, 
                  color = include, 
                  include_reason = selection_reason, 
                  who = requested_by, 
                  microcystis_percent = microcystis_abundance), inherit.aes = FALSE, size = 2.5) +
    scale_color_manual(values = c("green", "blue", "#edbb47","pink","#6c5a8f", "red", "#46874a")) +
  facet_grid(year ~ fct_relevel(NOAA_Site,sites_w_most_samples$NOAA_Site),scales = "free_x") +
  theme_bw() +
  labs(x = NULL, y= "log values") +
    theme(axis.text.y = element_blank(),
          axis.ticks.y = element_blank())
) %>% plotly::ggplotly()
```





